### PR TITLE
fix:[CDS-44134]:service-create-using-deployment-template

### PR DESCRIPTION
**Pull request by namanharness**
### Summary

- Creation of Service using Deployment template.
#### Screenshots

https://user-images.githubusercontent.com/110456765/193551255-de2e9fad-9fe5-412f-aa17-a0589cd079fc.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11623)

File: cypress/integration/72-templates-library/ServiceUsingDeploymentTemplate.spec.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        ]
+      }).as('enableFeatureFlag')
+    })
+
+    cy.initializeRoute()
+
+    cy.visit(servicesRoute, {
+      timeout: 30000
+    })
+    cy.wait(3000)
```
Comment (depanshu-dhiman) [inline cypress/integration/72-templates-library/ServiceUsingDeploymentTemplate.spec.ts:66]: Is this necessary after `cy.visit(servicesRoute, {
      timeout: 30000
    })`

**Review by depanshu-dhiman — COMMENTED** [review]
Please address

File: cypress/integration/72-templates-library/ServiceUsingDeploymentTemplate.spec.ts (hunk 1 above)
Comment (depanshu-dhiman) [reply cypress/integration/72-templates-library/ServiceUsingDeploymentTemplate.spec.ts:66]: Check `visitPageAssertion`

File: cypress/integration/72-templates-library/ServiceUsingDeploymentTemplate.spec.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  featureFlagsCall,
+  servicesRoute,
+  afterSaveServiceResponse,
+  afterSaveServiceNameResponse,
+  afterSaveServiceHeaderResponse
+} from '../../support/70-pipeline/constants'
+import { useTemplateResponse, afterUseTemplateListResponse } from '../../support/72-templates-library/constants'
+
+describe('ServiceV2 - Deployment Template', () => {
+  const deploymentTemplatesListCall =
```
[the remark this answers is not in the document]
Comment (namanharness) [reply cypress/integration/72-templates-library/ServiceUsingDeploymentTemplate.spec.ts:20]: sure 👍

File: cypress/integration/72-templates-library/ServiceUsingDeploymentTemplate.spec.ts (hunk 1 above)
Comment (namanharness) [reply cypress/integration/72-templates-library/ServiceUsingDeploymentTemplate.spec.ts:66]: Thanks for the tip, removed the `cy.wait(3000)`,
just doing it with  `visitPageAssertion` now

File: cypress/integration/72-templates-library/ServiceUsingDeploymentTemplate.spec.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    cy.contains('p', 'Custom').should('be.visible')
+    cy.contains('p', 'Artifactory').should('be.visible')
+    cy.contains('p', 'Jenkins').should('be.visible')
+    cy.contains('p', 'Nexus3').should('be.visible')
+    cy.contains('p', 'Amazon S3').should('be.visible')
+    cy.contains('p', 'Docker Registry').should('be.visible')
+    cy.contains('p', 'ECR').should('be.visible')
+    cy.contains('p', 'GCR').should('be.visible')
+    cy.contains('p', 'ACR').should('be.visible')
```
Comment (PratyushGar) [inline cypress/integration/72-templates-library/ServiceUsingDeploymentTemplate.spec.ts:98]: Maintain an array of artifactRepoTypes and iterate over it to assert the priority/order as well.

Comment (namanharness) [reply cypress/integration/72-templates-library/ServiceUsingDeploymentTemplate.spec.ts:98]: okay

**Review by depanshu-dhiman — APPROVED** [review]

**Pull request merged**